internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -61,54 +62,57 @@ func resourceComputeCluster() *pluginsdk.Resource {
 				ValidateFunc: validation.StringInSlice([]string{string(machinelearningservices.VMPriorityDedicated), string(machinelearningservices.VMPriorityLowPriority)}, false),
 			},
 
-			"identity": {
+			"identity": SystemAssignedUserAssigned{}.Schema(),
+
+			"scale_settings": {
 				Type:     pluginsdk.TypeList,
 				Required: true,
 				ForceNew: true,
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
-						"type": {
-							Type:     pluginsdk.TypeString,
+						"max_node_count": {
```

Comment:
This property is not introduced in this PR, so we can't change it.

Reply:
Could you please elaborate why we are unable to change the name?

Reply:
We can change the name, but in this PR.

Because this property is already released, so if it's really necessary to change the name, we must add a new property `max_instances` and deprecate the old one.

IMO, we can handle v1-to-v2 upgrade inside the tf resource, we don't have to expose them to end users, so they can have a smooth upgrade.